subprojects/core/src/main/java/org/gradle/internal/operations/trace/BuildOperationTrace.java

```diff
@@ -243,8 +258,12 @@ public boolean processLine(@SuppressWarnings("NullableProblems") String line) {
                     } else if (map.containsKey("time")) {
                         SerializedOperationProgress serialized = new SerializedOperationProgress(map);
                         PendingOperation pending = pendings.get(serialized.id);
-                        assert pending != null;
-                        pending.progress.add(serialized);
+                        if (pending != null) {
```

Comment:
TODO for me to fix this.

Reply:
this is currently blocking me quite a bit. I tried moving the BuildOperationTrace up to GlobalServices handling it similar as we do `BuildOperationNotificationBridge` and now `LoggingBuildOperationNotificationBridge`. But it doesn't capture the build operation events of nested projects. after explicitly registering `BuildOperationTrace#Listener` to `BuildOperationListenerManager` I could manage to redirect output of the nested build to the trace, but the build operation started / finished events didn't show up there. I've added an ignored test for this scenario at `LoggingBuildOperationNotificationIntegTest`  but the fixture is blowing due to the described problem